### PR TITLE
Pre and post services

### PR DESCRIPTION
**Pull request by fmherschel**
* The UT for sapservices-move should be a follow-up and could be introduced by a new PR
* The service names sapping and sappong are aligned with the needs of the customer proejct

### CODE REVIEW

**Review by arbulu89 — COMMENTED** [review]
@fmherschel Only 2 major comments.
- The fstring are dangeroues in our env, as they are not implemented in old python3 versions, and we might have this case
- The commented `sapservices` initialization looks suspicious

File: sbin/sapservices-move.in [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#
+
+import os
+import sys
+from pathlib import Path
+import argparse
+
+script_version = "2020-12-01 14:14"
+script_name = sys.argv[0]
+#sapservices = "/usr/sap/sapservices"
```
[the remark this answers is not in the document]
Comment (fmherschel) [reply sbin/sapservices-move.in:20]: No - my bad :) It was from a local test.

File: sbin/sapservices-move.in [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+sapservices = "./sapservices"
+
+def do_help():
+    print(f"usage: {script_name} [OPTION]")
```
[the remark this answers is not in the document]
Comment (fmherschel) [reply sbin/sapservices-move.in:24]: Already have commit on my side - will send soon.

File: sbin/sapservices-move.in [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    return 0
+
+def do_stat():
+    print(Path(sapservices).stat())
+
+def do_cat():
+    '''
+    document current sapservices content
+    '''
+    with open(sapservices) as infile:
```
[the remark this answers is not in the document]
Comment (fmherschel) [reply sbin/sapservices-move.in:38]: Yes your code is nice - I will adapt that. I have revied lot of options in the internet to find a good code but your one is shorter :) Will come soon. I will use print(f.read()) to be sure python3 does not rise an error.

File: sbin/sapservices-move.in [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    #
+    # only if sapservices is not already empty
+    #
+    if Path(sapservices).stat().st_size == 0:
+        print(f"skip moving {sapservices}")
+    else:
+        #
+        # touch empty file, move orig file, move-over empty file to hide
+        #
+        sapservices_empty = sapservices + ".empty"
```
[the remark this answers is not in the document]
Comment (fmherschel) [reply sbin/sapservices-move.in:57]: Yes sure I am open for such improvements in the future.
The best way would be to have an atom code for moving the file PLUS replacing by the empty file.

File: sbin/sapservices-move.in [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        do_stat()
+
+def do_unhide():
+    '''
+    unhide sapservices by restoring the previously hided version of the file
+    '''
+    do_stat()
+    do_cat()
+    if Path(sapservices).stat().st_size == 0:
+        sapservices_empty = sapservices + ".empty"
```
[the remark this answers is not in the document]
Comment (fmherschel) [reply sbin/sapservices-move.in:71]: ok maybe copy+paste and my vi did't warn me :)

**Review by arbulu89 — APPROVED** [review]
@fmherschel Thanks for the changes!
I think we are ready to merge. 
We will polish the code and add the tests next.

**Comment by fmherschel** [comment]
> @fmherschel Thanks for the changes!
> I think we are ready to merge.
> We will polish the code and add the tests next.

Yes thanks bag for you help! Very appreciated.

File: sbin/sapservices-move.in [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    do_stat()
+
+def main():
+    '''
+    main method - take only exact ONE parameter
+    '''
+    try:
+        main_param = sys.argv[1]
+        if main_param == "-v" or main_param == "--version":
+            print(f"version: {script_version}")
```
Comment (arbulu89) [inline sbin/sapservices-move.in:82]: When pending `fstring` here

File: sbin/sapservices-move.in [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    '''
+    unhide sapservices by restoring the previously hided version of the file
+    '''
+    do_stat()
+    do_cat()
+    if Path(sapservices).stat().st_size == 0:
+        sapservices_moved = sapservices + ".moved"
+        os.rename(sapservices_moved, sapservices) 
+    else:
+        print(f"skip unhiding {sapservices}")
```
Comment (arbulu89) [inline sbin/sapservices-move.in:72]: fstring

File: sbin/sapservices-move.in [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    '''
+    hide sapservices file by moving and creating an empty file 
+    '''
+    do_stat()
+    do_cat()
+    #
+    # only if sapservices is not already empty
+    #
+    if Path(sapservices).stat().st_size == 0:
+        print(f"skip moving {sapservices}")
```
Comment (arbulu89) [inline sbin/sapservices-move.in:50]: fstring

**Pull request merged**